qiskit-superstaq/qiskit_superstaq/superstaq_backend.py

```diff
@@ -136,3 +206,234 @@ def run(
         job = qss.SuperstaQJob(self, job_id)
 
         return job
+
+    def compile(
+        self,
+        circuits: Union[qiskit.QuantumCircuit, List[qiskit.QuantumCircuit]],
+        **kwargs: Any,
+    ) -> qss.compiler_output.CompilerOutput:
+        """Compiles the given circuit(s) to the backend's native gateset.
+
+        Args:
+            circuits: The qiskit QuantumCircuit(s) to compile.
+            kwargs: Other desired compile options.
+
+        Returns:
+            A CompilerOutput object whose .circuit(s) attribute contains optimized compiled
+            circuit(s).
+
+        Raises:
+            ValueError: If `target` is not a valid AQT or IBMQ target.
+        """
+        _validate_qiskit_circuits(circuits)
+        if self.name().startswith("ibmq_"):
+            return self.ibmq_compile(circuits, **kwargs)
+        elif self.name().startswith("aqt_"):
+            return self.aqt_compile(circuits, **kwargs)
+        elif self.name().startswith("sandia_"):
+            return self.qscout_compile(circuits, **kwargs)
+        elif self.name().startswith("cq_"):
+            return self.cq_compile(circuits, **kwargs)
+        raise ValueError(
+            f"{self.name()} is not a valid target (currently supports AQT, IBMQ, QSCOUT, CQ)."
+        )
+
+    def aqt_compile(
+        self,
+        circuits: Union[qiskit.QuantumCircuit, List[qiskit.QuantumCircuit]],
+        num_equivalent_circuits: Optional[int] = None,
+        random_seed: Optional[int] = None,
+        atol: Optional[float] = None,
+        **kwargs: Any,
+    ) -> qss.compiler_output.CompilerOutput:
+        """Compiles and optimizes the given circuit(s) for the Advanced Quantum Testbed (AQT) at
+        Lawrence Berkeley National Laboratory. Also allows using Equivalent Circuit Averaging (ECA).
+
+        See arxiv.org/pdf/2111.04572.pdf for a description of ECA.
+
+        Args:
+            circuits: The qiskit QuantumCircuit(s) to compile.
+            num_equivalent_circuits: Number of logically equivalent random circuits to generate for
+                each input circuit.
+            random_seed: Optional seed for circuit randomizer.
+            atol: Tolerance to use for approximate gate synthesis (currently just for qutrit gates).
+            kwargs: Other desired compile options.
+
+        Returns:
+            Object whose .circuit(s) attribute contains the optimized circuits(s). Alternatively for
+            ECA, an Object whose .circuits attribute is a list (or list of lists) of logically
+            equivalent circuits If qtrl is installed, the object's .seq attribute is a qtrl Sequence
+            object containing pulse sequences for each compiled circuit, and its .pulse_list(s)
+            attribute contains the corresponding list(s) of cycles.
+
+        Raises:
+            ValueError: If `target` is not a valid AQT target.
```

Comment:
should be something like
```suggestion
            ValueError: If this is not an AQT backend.
```
(ditto for the other `xx_compile`s)

Reply:
(Others done as well.)